agent/__init__.py

```diff
@@ -404,14 +406,68 @@ def renew_expired_cert(csr, device_id, debug=False):
     }
 
 
-def run(ping=True, debug=False, dev=False):
+def setup_endpoints(dev):
     if dev:
         global WOTT_ENDPOINT, MTLS_ENDPOINT, DASH_ENDPOINT
         endpoint = os.getenv('WOTT_ENDPOINT', 'http://localhost')
         DASH_ENDPOINT = endpoint + ':' + str(DASH_DEV_PORT)
         WOTT_ENDPOINT = endpoint + ':' + str(WOTT_DEV_PORT) + '/api'
         MTLS_ENDPOINT = endpoint + ':' + str(MTLS_DEV_PORT) + '/api'
 
+
+def fetch_creds(debug, dev):
+
+    print('Fetching credentials...')
+    can_read_cert()
+
+    creds_req = requests.get(
+        '{}/v0.2/creds'.format(MTLS_ENDPOINT),
+        cert=(CLIENT_CERT_PATH, CLIENT_KEY_PATH),
+        headers={
+            'SSL-CLIENT-SUBJECT-DN': 'CN=' + get_device_id(),
+            'SSL-CLIENT-VERIFY': 'SUCCESS'
+        } if dev else {}
+    )
+    if not creds_req.ok:
+        print('Fetching failed.')
+        if debug:
+            print("[RECEIVED] Fetch creds: code {}".format(creds_req.status_code))
+            print("[RECEIVED] Fetch creds: {}".format(creds_req.content))
+        return
+    creds = creds_req.json()
+
+    print('Credentials retreived.')
+    if debug:
+        print('Creds: {}'.format(creds))
+
+    by_names = {}
```

Comment:
What does 'by_name" mean?

Reply:
the credentials was got in request are in array, so to make only one writes for each file resort it to dictionary by their 'name'...so may be not actually right var name